reporter/v2/pkg/reporter/reconcile_task.go

```diff
@@ -158,7 +219,7 @@ func (r *ReconcileTask) ReportTask(ctx context.Context, report *marketplacev1alp
 }
 
 func (r *ReconcileTask) CanRunUploadTask(ctx context.Context) bool {
```

Comment:
CanRunUploadTask should be true if you can do it. If you return just `IsDisconnected` then that would flip this. Try inversing the value.